tools/generator/src/Generator+DistinguishTargets.swift

```diff
@@ -0,0 +1,257 @@
+import CryptoKit
+import XcodeProj
+
+/// A `struct` containing values for a target that need to be unique in the eyes
+/// of Xcode, as well as the target itself.
+///
+/// Xcode requires that certain properties of targets are unique, such as name,
+/// the `TARGET_NAME` and `PRODUCT_MODULE_NAME` build settings, etc. This class
+/// collects information on all of the targets and calculates unique values for
+/// each one. If the values are user facing (i.e. the target name), then they
+/// are formatted in a readable way.
+struct DistinguishedTarget: Equatable {
```

Comment:
I had a hard time coming up with a name for this. I think `DistinguishedTarget` is _fine_, but should it be something else like `DisambiguatedTarget`?

Reply:
Yeah both are kinda wordy. Disambiguated is at closer to what you I think this represents.

Naming is hard!